library/src/main/java/bio/terra/landingzone/stairway/flight/create/resource/step/CreateAksStep.java

```diff
@@ -215,6 +236,47 @@ private static void enableWorkloadIdentity(KubernetesCluster aks) {
     aks.refresh();
   }
 
+  private void enableCostSavings(
+      KubernetesCluster aks,
+      String vNetId,
+      boolean costSavingsEnabled,
+      boolean autoScalingEnabled) {
+    // Enable a spot nodepool if cost savings are enabled. TODO: VPA
+    if (costSavingsEnabled) {
+      var aksPartialUpdate =
+          aks.update()
+              .defineAgentPool(SPOT_NODE_POOL_NAME)
+              .withVirtualMachineSize(
+                  ContainerServiceVMSizeTypes.fromString(
+                      parametersResolver.getValue(
+                          CromwellBaseResourcesFactory.ParametersNames.AKS_MACHINE_TYPE.name())))
+              .withAgentPoolVirtualMachineCount(
+                  Integer.parseInt(
+                      parametersResolver.getValue(
+                          CromwellBaseResourcesFactory.ParametersNames.AKS_NODE_COUNT.name())))
```

Comment:
@rtitle when adding spot nodes as an optional addition to LZ, should we default it to the same node count as AKS default? Or should we do something different here, i.e. make this higher?

Reply:
I think the ideal would be to have separate parameters for machine type and node count for the spot node pool - if those parameters are not supplied by the caller, they could default to the machine type/node count used by the system pool.

Reply:
I did a bit more digging on this and here is what I found out: 
- number of nodes per pool is constrained by IPs in the provided subnet
- the way LZ is current set up, we can go up to 3 nodes
- it is not recommended to do spot nodes without auto scaling enabled since it could be possible that when all nodes are evicted, the count goes down to 0 and then someone manually would need to get it back to 1 to make it be able to have pods run on it

with that I decided:
- we will keep the spot node overall count the same as the system aks node
- we will allow to change the machine type of the spot node but if none is provided, default to the system machine type
- we will not give an option to disable auto scale on spot nodes
- we will allow users to set a max scale machine number, but default will be 10